packages/core/src/RushTool.ts

```diff
@@ -0,0 +1,64 @@
+import path from "path";
+import fs from "fs-extra";
+import jju from "jju";
+
+import { PackageJSON } from "@changesets/types";
+import { Tool, ToolType, Package, Packages, MonorepoRoot } from './Tool';
+
+interface RushJson {
+    projects: RushProject[];
+}
+
+interface RushProject {
+    packageName: string;
+    projectFolder: string;
+}
+
+export const RushTool: Tool = {
+    type(): ToolType {
+        return 'rush';
+    },
+
+    async isMonorepoRoot(directory: string): Promise<MonorepoRoot | undefined> {
+        try {
+            await fs.readFile(path.join(directory, "rush.json"), "utf8");
+            return {
+                dir: directory,
+                tool: RushTool
+            };
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+    },
+
+    async getPackages(directory: string): Promise<Packages | undefined> {
+        try {
+            const rushText: string = await fs.readFile(path.join(directory, "rush.json"), "utf8");
+
+            // Rush configuration files are full of inline and block-scope comment blocks (JSONC),
+            // so we use a parser that can handle that.
+            const rushJson: RushJson = jju.parse(rushText);
+
+            const directories = rushJson.projects.map(project => path.resolve(directory, project.projectFolder)).sort();
+            const packageJsons = await Promise.all(directories.map(dir => fs.readJson(path.join(dir, "package.json"))));
```

Comment:
So you don't support globs to catch all packages in a given dir? 🤔 Is this list somehow auto-updated by Rush itself or it is always the responsibility of the user to add new packages to the declared list?

Reply:
Rush tends to favor explicit configuration over convention, and doesn't support globs for that reason.

A nice side effect of this approach is that you can guarantee that in order to build and test a new project, the rush.json file must be touched, so your maintainers team can be a codeowners of that file if you want them to approve the initial skeleton PRs for new projects.